crates/task-impls/src/consensus.rs

```diff
@@ -1319,7 +1319,7 @@ where
                 view_number: view,
                 justify_qc: consensus.high_qc.clone(),
                 parent_commitment: parent_leaf.commit(),
-                block_header: TYPES::BlockHeader::new(*payload_commitment, &parent_header),
+                block_header: TYPES::BlockHeader::new(*payload_commitment, (), &parent_header),
```

Comment:
I'm confused how we can pass `()` here for the metadata, when we don't know that `TYPES::BlockHeader::Metadata = ()`. We should really be getting the metadata from whoever built the block, ie wherever `self.payload_commitment` came from, we should get `self.payload_metadata` from the same place.